<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1270</width>
    <height>940</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1270</width>
    <height>940</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1270</width>
    <height>940</height>
   </size>
  </property>
  <property name="font">
   <font>
    <family>Segoe UI Emoji</family>
    <pointsize>9</pointsize>
   </font>
  </property>
  <property name="windowTitle">
   <string>Tree visualization</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>90</y>
      <width>1051</width>
      <height>851</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <property name="spacing">
      <number>2</number>
     </property>
     <item>
      <widget class="betterGraphicsView" name="graphicsView">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>809</width>
         <height>790</height>
        </size>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout" stretch="0,0,0,0">
       <property name="spacing">
        <number>1</number>
       </property>
       <item>
        <widget class="QPushButton" name="AVL_button">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="font">
          <font>
           <family>Segoe UI Emoji</family>
           <pointsize>9</pointsize>
          </font>
         </property>
         <property name="text">
          <string>AVL-tree</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="RB_button">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>50</width>
           <height>0</height>
          </size>
         </property>
         <property name="font">
          <font>
           <family>Segoe UI Emoji</family>
           <pointsize>9</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Red-Black tree</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="Treap_button">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="font">
          <font>
           <family>Segoe UI Emoji</family>
           <pointsize>9</pointsize>
           <weight>50</weight>
           <bold>false</bold>
          </font>
         </property>
         <property name="text">
          <string>Treap</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="Splay_button">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="font">
          <font>
           <family>Segoe UI Emoji</family>
           <pointsize>9</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Splay-tree</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
   <widget class="QLabel" name="current_tree_label">
    <property name="geometry">
     <rect>
      <x>390</x>
      <y>10</y>
      <width>321</width>
      <height>61</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Segoe UI Semilight</family>
      <pointsize>28</pointsize>
     </font>
    </property>
    <property name="text">
     <string>AVL-Tree</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>1080</x>
      <y>130</y>
      <width>171</width>
      <height>341</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_4">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout">
       <property name="sizeConstraint">
        <enum>QLayout::SetFixedSize</enum>
       </property>
       <item>
        <widget class="QLineEdit" name="num_add_edit"/>
       </item>
       <item>
        <widget class="QPushButton" name="add_button">
         <property name="font">
          <font>
           <family>Segoe UI Emoji</family>
           <pointsize>9</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Insert node</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="checkBox">
         <property name="font">
          <font>
           <family>Segoe UI Emoji</family>
           <pointsize>9</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Remove mode</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="clear_button">
         <property name="font">
          <font>
           <family>Segoe UI Emoji</family>
           <pointsize>9</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Clear</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <spacer name="verticalSpacer">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QLabel" name="label">
           <property name="font">
            <font>
             <family>Segoe UI Emoji</family>
             <pointsize>9</pointsize>
            </font>
           </property>
           <property name="text">
            <string>Insert</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="nodesCount"/>
         </item>
         <item>
          <widget class="QLabel" name="label_2">
           <property name="font">
            <font>
             <family>Segoe UI Emoji</family>
             <pointsize>9</pointsize>
            </font>
           </property>
           <property name="text">
            <string>nodes</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QLabel" name="label_3">
           <property name="font">
            <font>
             <family>Segoe UI Emoji</family>
             <pointsize>9</pointsize>
            </font>
           </property>
           <property name="text">
            <string>from</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="nodesMin">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maximumSize">
            <size>
             <width>120</width>
             <height>16777215</height>
            </size>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <widget class="QLabel" name="label_4">
           <property name="font">
            <font>
             <family>Segoe UI Emoji</family>
             <pointsize>9</pointsize>
            </font>
           </property>
           <property name="text">
            <string>to</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="nodesMax">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maximumSize">
            <size>
             <width>120</width>
             <height>16777215</height>
            </size>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QPushButton" name="insertRandom">
         <property name="text">
          <string>Insert</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>betterGraphicsView</class>
   <extends>QGraphicsView</extends>
   <header>bettergraphicsview.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
